src/react/OCConnect.js

```diff
@@ -37,7 +39,7 @@ const OCConnect = ( { children, opts, sandboxMode } ) =>
         } );
     };
 
-    return <OCContext.Provider value={ { ocAuth, authState, updateAuthState } }>{ children }</OCContext.Provider>;
+    return <OCContext.Provider value={ { OCId, ethAddress, ocAuth, authState, updateAuthState, setOCId, setEthAddress } }>{ children }</OCContext.Provider>;
```

Comment:
I think we should just set the states in line#37 and not returning the setters. I meant the child component is not supposed to change that state, afterall. They are pure consumer.

Reply:
actually we don't even need to keep OCId and ethAddress as specific state inside here.

would something like this works?

```
return <OCContext.Provider value={ { OCId: ocAuth.OCId(), ethAddress: ocAuth.ethAddress(), ocAuth, authState, updateAuthState} }>{ children }</OCContext.Provider>;
```